src/selectors/getChildren.ts

```diff
@@ -73,4 +81,32 @@ export const getChildrenRanked = (state: State, context: Context): Child[] =>
 
 /** Returns the first visible child of a context. */
 export const firstVisibleChild = (state: State, context: Context) =>
-  getChildrenSorted(state, context)[0]
+  getVisibleChildrenSorted(state, context)[0]
+
+/**
+ * Returns only visible thoughts from the children array. It checks if a thought lies within the cursor path.
+ *
+ * @param state - Redux state.
+ * @param path - Parent's path.
+ * @param simplePath - Parent's simple path.
+ * @param children - Children of the given simplePath's context.
+ */
+export const getVisibleChildrenWithCursorCheck = (state: State, path: Path, simplePath: SimplePath, children: (Child | ThoughtContext)[]) => {
+
+  const showContexts = isContextViewActive(state, pathToContext(path))
+
+  return children.filter(child => {
+
+    const childSimplePath = getChildPath(state, child, simplePath, showContexts)
+    const childPath = appendChildPath(state, childSimplePath, path)
+
+    const value = showContexts
+      ? head((child as ThoughtContext).context)
+      : (child as Child).value
```

Comment:
If you'd like a little challenge, you can try to refactor the filter predicate as a composition ;)

```json
or(showHiddenThoughts, or(isChildVisible, isChildInCursor))
```